src/snowflake/snowpark/session.py

```diff
@@ -1658,9 +1658,53 @@ def query_history(self) -> QueryHistory:
         return query_listener
 
     def _table_exists(self, table_name: str):
-        # TODO: Support qualified table names
-        tables = self._run_query(f"show tables like '{table_name}'")
-        return tables is not None and len(tables) > 0
+        # implementation based upon: https://docs.snowflake.com/en/sql-reference/name-resolution.html
+        validate_object_name(table_name)
+        try:
+            qualified_table_name = table_name.split(".")
+            if len(qualified_table_name) == 1:
+                # name in the form of "table"
+                tables = self._run_query(f"show tables like '{table_name}'")
+            elif len(qualified_table_name) == 2:
+                # name in the form of "schema.table" omitting database
+                # schema: qualified_table_name[0]
+                # table: qualified_table_name[1]
+                tables = self._run_query(
+                    f"show tables like '{qualified_table_name[1]}' in schema {qualified_table_name[0]}"
+                )
+            else:  # len(qualified_table_name) == 3, there is no other case, validate_object_name checked for us
+                # name in the form of "database.schema.table"
+                # database: qualified_table_name[0]
+                # schema: qualified_table_name[1]
+                # table: qualified_table_name[2]
+                if qualified_table_name[1] == "":
+                    # Double-Dot Notation: "database..table" in which the schema is empty
+                    tables = self._run_query(
+                        f"show tables like '{qualified_table_name[2]}' in database {qualified_table_name[0]}"
+                    )
+                else:
+                    # "database.schema.table", schema is not empty
+                    tables = self._run_query(
+                        f"show tables like '{qualified_table_name[2]}' in database {qualified_table_name[0]}"
+                    )
+                    # table is a tuple and the 3rd col(0 based idx) is schema_name
+                    return any(
+                        [
+                            table
+                            for table in tables
+                            if table[3].lower() == qualified_table_name[1].lower()
+                        ]
+                    )
+
+            return tables is not None and len(tables) > 0
+        except ProgrammingError as exc:
+            if exc.errno == 2043:
+                # SQL compilation error: Object does not exist, or operation cannot be performed
+                return False
```

Comment:
When does this happen? Is it because the database/schema does not exist?

It would be nice to make 2043 a static constant. Hopefully the same as the variable name in GS (if there is one).

Reply:
yes, this happens when database/schema does not exist.

actually after reconsidering the case, I feel this case we shall just raise the error instead of returning false..
returning `False` is kinda misleading because it's telling users that the `table` doesn't exist, not the schema or database.

If such error happens, users should go and check their databse/schema.

thoughts?

Reply:
I also think we should just raise an exception here

Reply:
When a database/schema does not exist, I think the table creation should fail, right? If that's the case, then it seems legit to just raise an exception because the following table creation will fail on the server anyways.

Reply:
yes, if no db/schema, then we would get schema/db not exist error.